### PR TITLE
Feature/deployment setup

### PR DESCRIPTION
**Pull request by amitsha2**

### CODE REVIEW

File: deployment/Dockerfile.prod [hunk 1]
```diff
@@ -0,0 +1,4 @@
+FROM segment/chamber:2 AS chamber
```
Comment (altonotch) [inline deployment/Dockerfile.prod]: I don't like this as it:

a) changes base image. Is it alpine? Is it debian? 
b) we don't know the exact version installed.

File: deployment/Dockerfile.prod [hunk 2]
```diff
@@ -0,0 +1,4 @@
...
+FROM base_image
+
+COPY --from=chamber /chamber /bin/chamber
```
Comment (altonotch) [inline deployment/Dockerfile.prod]: I suggest to move the cmd here. This is spreading too much at this point.

File: deployment/docker-compose.template.yml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+version: '3'
+services:
+  api:
+    image: ${AWS_ACCOUNT_ID}.dkr.ecr.${AWS_REGION}.amazonaws.com/peerbook/server:${VERSION}
+    command: sh -c "/bin/chamber exec ${LAYER_NAME} -- /bin/myapp"
```
Comment (altonotch) [inline deployment/docker-compose.template.yml]: Move to dockerfile.prod. 
Also, I think it's `/app/server` and not `/bin/myapp`.

File: deployment/docker-compose.template.yml [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+    hostname: api
+    ports:
+      - "17777"
+    environment:
+      AWS_REGION: ${AWS_REGION}
```
Comment (altonotch) [inline deployment/docker-compose.template.yml:9]: Inject the `layer_name` here.

File: deployment/docker-compose.template.yml [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+    logging:
+      driver: awslogs
+      options:
+        awslogs-group: api
```
Comment (altonotch) [inline deployment/docker-compose.template.yml]: `staging-peerbook-api`. CloudWatch is per region.

File: deployment/docker-compose.template.yml [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+    image: ${AWS_ACCOUNT_ID}.dkr.ecr.${AWS_REGION}.amazonaws.com/peerbook/server:${VERSION}
+    command: sh -c "/bin/chamber exec ${LAYER_NAME} -- /bin/myapp"
+    hostname: api
+    ports:
+      - "17777"
+    environment:
+      AWS_REGION: ${AWS_REGION}
+    logging:
+      driver: awslogs
+      options:
```
Comment (altonotch) [inline deployment/docker-compose.template.yml]: Add `awslogs-stream-prefix`

File: deployment/ecs-params.template.yml [hunk 7]
```diff
@@ -0,0 +1,13 @@
+version: 1
+task_definition:
+  task_execution_role: arn:aws:iam::${AWS_ACCOUNT_ID}:role/ecs/${ENVIRONMENT}_peerbook_ecs_task_executor
+  services:
+    api:
+      essential: true
+      cpu_shares: 200
+      mem_limit: "512m"
+      healthcheck:
+        test: [ 'CMD', 'curl', '-s', '-o', '/dev/null', '-w', '"%{http_code}"', 'http://localhost:17777/health/check/' ]
```
Comment (altonotch) [inline deployment/ecs-params.template.yml]: They don't have yet a heath check endpoint, ping `"/"` instead.

File: deployment/docker-compose.template.yml [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+        awslogs-group: api
+        awslogs-region: ${AWS_REGION}
```
Comment (altonotch) [inline deployment/docker-compose.template.yml]: Add blank line after this line.

File: deployment/ecs-params.template.yml [hunk 9]
```diff
@@ -0,0 +1,13 @@
...
+        start_period: 5s
+        interval: 10s
+        retries: 5
```
Comment (altonotch) [inline deployment/ecs-params.template.yml]: Add blank line after this line

File: .github/workflows/main.yml [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+      - uses: actions/checkout@v2
+
+      # Runs a single command using the runners shell
+      - name: build and push
+        env: # Or as an environment variable
+          AWS_ACCOUNT_ID: ${{ secrets.AWS_ACCOUNT_ID }}    
+          AWS_REGION: ${{ secrets.AWS_REGION }}      
+        run: |
+          cd $GITHUB_WORKSPACE
+          bash deployment/build_and_push.sh
```
Comment (altonotch) [inline .github/workflows/main.yml:32]: How did you ensure that this step has aws cli v2?

File: .github/workflows/main.yml [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+
+
+      # Runs a set of commands using the runners shell
+      - name: deploy
+        env: # Or as an environment variable
+          AWS_ACCOUNT_ID: ${{ secrets.AWS_ACCOUNT_ID }}    
+          AWS_REGION: ${{ secrets.AWS_REGION }} 
+          LAYER_NAME: ${{ secrets.LAYER_NAME }}
+          TARGET_GROUP_ID: ${{ secrets.TARGET_GROUP_ID }}
+        run: bash deployment/deploy.sh
```
Comment (altonotch) [inline .github/workflows/main.yml:44]: How did you ensure that this step has aws cli v2 and ecs-cli?

File: deployment/build_and_push.sh [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+VERSION=$(git rev-parse --short HEAD)
+ECR_HOST="${AWS_ACCOUNT_ID}.dkr.ecr.${AWS_REGION}.amazonaws.com"
+ECR_URL="${ECR_HOST}/peerbook"
+
+if [ -z "${AWS_REGION+unset}" ]; then
+    echo "AWS_REGION must be set to push the image"
+    exit 1
+else
+    echo "Login into ECR"
+    aws ecr get-login-password --region "${AWS_REGION}" | docker login --username AWS --password-stdin "${ECR_HOST}"
```
Comment (altonotch) [inline deployment/build_and_push.sh:13]: Are you sure if this fails, the job exits here?

**Review by altonotch — CHANGES_REQUESTED** [review]

File: .github/workflows/main.yml (hunk 11 above)
Comment (amitsha2) [reply .github/workflows/main.yml:44]: from what i can tell the ubuntu runner has aws cli
added installation for ecs-cli

File: deployment/build_and_push.sh (hunk 12 above)
Comment (amitsha2) [reply deployment/build_and_push.sh:13]: yes

**Pull request merged**